Add time series module

sophieberkhout opened the pull request:
This module contains four analyses:

- Descriptives
- Stationarity
- ARIMA
- Spectral analysis

The idea is that the BSTS and Prophet will be merged into this module as separate analyses (https://github.com/jasp-stats/INTERNAL-jasp/issues/2128). Not sure what the best way is to do that.

Features that are not yet implemented but I will add later:

- (S)ARIMA data simulation
- Insights into the auto.arima. It is not easy to extract what models the auto.arima has searched and what the AIC/BICs were, unfortunately. I will see if I can find a way to extract this or add an option to manually specify which models to search (which would probably be a lot slower, but also more transparent). I would then also add table and plot of the (weighted) AIC/BICs.

sophieberkhout commented:
For testing, you can use the prophet dataset in the data library. I also made a jasp file for the arima analysis with a different dataset (will add that to the data library).
[Dutch House Sales.zip](https://github.com/jasp-stats/jaspTimeSeries/files/11904223/Dutch.House.Sales.zip)

Kucharssim commented on R/ARIMATimeSeries.R:
```diff
@@ -0,0 +1,642 @@
+
+  # Check if ready
+  if (!ready) {
+    rows <- data.frame(
+      sigma = ".",
+      ll = ".",
+      aicc = ".",
+      aic = ".",
+      bic = "."
+    )
+    row.names(rows) <- paste0("row", 1)
+    table$addRows(rows)
```
You don't have to do this, if the jaspTable is empty it should be already formatted this way.

Kucharssim reviewed (commented):
Looks very good!

I found a couple of bugs in stationarity, see  [time-series-bugs.zip](https://github.com/jasp-stats/jaspTimeSeries/files/11915561/time-series-bugs.zip) (rename .zip to .jasp):

- Manual box-cox transformation crashes:
![Screenshot 2023-06-30 at 12 43 31](https://github.com/jasp-stats/jaspTimeSeries/assets/7093546/259e9683-e2b9-45e3-b3c1-16129d7d15fd)

- Selecting density in the time-series plot crashes:
![Screenshot 2023-06-30 at 12 43 37](https://github.com/jasp-stats/jaspTimeSeries/assets/7093546/0540c9c3-7f93-4b45-b9db-85293198bb86)


I wonder if it makes sense to include the two parameter box-cox transformation, not only the one parameter box-cox transformation - especially since we offer the two parameter version in jaspBase as well (https://github.com/jasp-stats/jaspBase/blob/master/R/transformFunctions.R).

sophieberkhout commented:
Thank you, Simon!

I fixed the bugs and removed the unnecessary empty table formatting (I only kept it for the stationarity tests table as I like that it shows the name of the tests and the null hypotheses).

Regarding the Box-Cox transformation, I see what you mean. The forecast package actually uses a modification of the Box-Cox transformation so that it can handle negative values:
<img width="566" alt="image" src="https://github.com/jasp-stats/jaspTimeSeries/assets/49243293/662a482c-d52e-4c71-8004-6d80f12de1f9">

As far as I am aware, the shift parameter, like in jaspBase, is only used to make the data positive so that the Box-Cox transformation can be applied. The shift should be equal or greater than the lowest value in the data.

So they are two different methods to handle negative values. What do you think makes sense here? Use the jaspBase Box-Cox with shift parameter to be consistent with JASP? Or use the forecast Box-Cox so users do not have to think about a shift parameter?

Kucharssim reviewed (approved):
> The forecast package actually uses a modification of the Box-Cox transformation so that it can handle negative values:

Ahhh ok, then it's fine. My point was mainly to make sure that we can handle negative values.

> So they are two different methods to handle negative values. What do you think makes sense here? Use the jaspBase Box-Cox with shift parameter to be consistent with JASP? Or use the forecast Box-Cox so users do not have to think about a shift parameter?

I think we can leave it as is for now. If anything then it's probably best to implement this modification of the box-cox in jaspBase as well.

> The idea is that the BSTS and Prophet will be merged into this module as separate analyses.

I can make a separate PR for that.

The current functionality is very nice, pretty cool!
